feat: migrate to new project layout

napei opened the pull request:
From #360 

This PR moves things around a lot, with the end goal to rewrite everything in TypeScript, with nice modules and type safety we well as re-write all tests and involved logic to a more modern approach (separate task).

ypid commented:
If you want to continue this, I would suggest to not make further changes but get all tests passing so that I can just merge it. Otherwise this big refactor could get out of sync and rebasing it would be a pain.

napei commented:
@ypid I made some edits to get tests working. I was going through all the diff checks and extra parsing checks, but I got lost.

There are 207 tests from `test/test.js` not passing. No idea why and me trying to figure out is pointless so I might as well throw it to you to have a look at 😄 Feel free to checkout my branch and have a look.

I'm not familiar with Make at all so I'm a bit biased and want to get rid of it. But it _was_ working so hopefully it's a simple fix. Just not in my bandwidth at the moment.

ypid commented:
GNU Make is still awesome, even with various attempts at replacing it in Python, node.js or so. Only thing I have not tried is the build system of Rust. If it helps, you can ignore the Makefile. I know I made it kinda complex. The idea of the test framework is that `node test.js --library-file opening_hours.min.js > test.en.log 2>&1` should produce the exact same `test.en.log` as it is already tracked in git. If that is true then the biggest part of the tests suite passes. That is what the Makefile tests. The (as I think) cleaver part is that you can make large scale changes (which change half of the test outputs) and review and accept it simply by one `git add` instead of updating all the unit tests with your editor. Show me one test framework that can do that.

Thanks for giving it a try! I hope it will be easy. I cannot look into it right now but hopefully some day.

ypid commented:
I got it working. Can you have a final look. I know there are still broken things in there but I would rather merge this now.

napei commented on package.json:
```diff
@@ -34,36 +34,44 @@
   },
   "scripts": {
-    "build": "make build",
+    "build": "rollup -c",
```
Probably should change this back to `make`. My bad.

ypid replied:
Sounds good. Done.

ypid commented on Makefile:
```diff
@@ -522,19 +519,17 @@ osm-tag-data-gen-stats-sort:
-
 %.min.js: %.js
-	./node_modules/.bin/terser --output "$@" --comments '/github.com/' "$<"
+	node_modules/.bin/esbuild --output "$@" --comments '/github.com/' "$<"
```
Fixed.

esbuild is faster than the typical JavaScript tools (esbuild is in golang). So I would like to use it. But I could not get it to work. If you can help, let me know or push a commit to your branch.

napei replied:
I had esbuild working fine inside rollup generating assets. Double check that method and I'll have a look when I get the chance.

ypid replied:
I saw it worked together with rollup. But I would like to use the Makefile for that, should be possible.

Note: I just rechecked my research on build systems and still, nothing fully replaces GNU make for me. For smaller use cases, it is fast and generic.

ypid commented:
Tests are passing now. Some more work is needed, for example the README and the deployment. Can we make a deal @napei, I make sure I can deploy to https://openingh.openstreetmap.de/evaluation_tool/ without breaking anything and you update the README? I will keep the URL locations the same for now.

Also, before I merge this, could a third person look over this?

napei commented:
> Tests are passing now. Some more work is needed, for example the README and the deployment. Can we make a deal @napei, I make sure I can deploy to https://openingh.openstreetmap.de/evaluation_tool/ without breaking anything and you update the README? I will keep the URL locations the same for now.
> 
> Also, before I merge this, could a third person look over this?

Alright. What specifically needs to change in the readme?

ypid commented:
> Alright. What specifically needs to change in the readme?

If I would know the specifics, I would have changed it already. I just saw that `make opening_hours+deps.min.js` is no longer up-to-date so I imaging other things might be outdated by this as well.

napei commented:
> > Alright. What specifically needs to change in the readme?
> 
> If I would know the specifics, I would have changed it already. I just saw that `make opening_hours+deps.min.js` is no longer up-to-date so other things might be outdated by this as well.

@ypid I corrected a few references to scripts and tests and in general the new directory structure. I also ran it against `markdownlint` and included that as an npm script as well as `doctoc` for the TOC. You could make these into a make script or even part of the pre-commit hook if you wanted to. 90% of my changes are formatting-related but given the tooling in place it's now consistent and automated.

I've added prettier as well to enforce consistency, perhaps running that against the entire repo to format code wouldn't be a bad idea which I'm happy to set up and commit if you want, and then enforcing it as a pre-commit hook which I've done in other repos. I love using it, it's always been the best and most correct formatter. Let me know what you think.

Given you've changed to pre-commit, which requires `pip install` (in a javascript library....) I am wondering if those dependencies need to be documented. Husky was working fine, and integrated with npm in a way that anyone cloning the repo and running `npm i` would have the hooks installed and working fine. Now they also have to pip install and run a command. Let me know what you think.

ypid commented on README.md:
````diff
@@ -122,553 +121,587 @@ around-the-clock shop with some breaks.
-```Shell
+```text
 git clone --recursive https://github.com/opening-hours/opening_hours.js.git
 ```
````
Clearly a shell command, same below :)

napei replied:
`Shell` isn't a language supported by GitHub. Your options are `sh, shell-script, bash or zsh` https://github.com/github/linguist/blob/1c106773c7496fa23e1a2dd04f27aa9be4ddff3c/lib/linguist/languages.yml#L5194-L5202

Spoler alert they all look the same as text

text
```text
git clone --recursive https://github.com/opening-hours/opening_hours.js.git
```

sh
```sh
git clone --recursive https://github.com/opening-hours/opening_hours.js.git
```

For semantics sake, I'll make them `sh`.

napei replied:
Fixed in db3ee9c

ypid reviewed (changes requested):
I was more interested in content fixes than style fixes. They should be discussed first. I am fine with using `-` for bullet items. But the

```Markdown
_date_
```

for emphasizing I cannot wrap my head around. If you want a standard, I suggest to use the first option from https://spec.commonmark.org/0.29/ if there are multiple. It just so happens to be `-` (bullet item) and `*` emphasizing.

Why make the link destinations lower case?

You don’t really need the pre-commit right now and I would rather have something universal that I can also use in other projects.

napei commented:
> But the
> 
> ```gfm
> _date_
> ```
> 
> for emphasizing I cannot wrap my head around.

@ypid I understand. I was following the style that GitHub (where this readme is published) uses.
![image](https://user-images.githubusercontent.com/8946502/107299302-22d19d00-6acb-11eb-8469-99e5d09355a4.png)

Happy to abandon those changes if you wish.

> Why make the link destinations lower case?

That only applies to shortcut keys using the `[key]: link` syntax because semantically `[text][Key]` and `[text][key]` are different. Even though it's supported that way it makes it consistent, which is the point.

ypid commented:
> I was following the style that GitHub

Seems GitHub is not very consistent here, ref: https://guides.github.com/features/mastering-markdown/

I would say change it back.

> Not necessarily a lot of effort but still, worth considering.

Point for you. OK, you can revert my commit with pre-commit if you want to.

napei commented:
@ypid reverted to use husky, and changed the readme to use `* *` for emphasis now instead of `_ _`. 

Also switched rollup plugins to `@rollup/plugin-...` packages as the older `rollup-plugin-...` ones are deprecated. Everything still compiles fine.

On the topic of pre-commits I changed `make check` in the pre-commit to `npm run test` which in turn runs `make check-full`. Makes more sense to keep everything inside `npm` scripts then it can more easily be changed later on. For me though, `make check` was spitting out errors around no targets for `check-diff` so if you want to have a look at that as I'm not sure what that's supposed to be doing.

`make check` also opens what I think is `vi` or something similar with a whole lot of `~`s down the terminal and I have to press `q` to exit it, and then presented with this

![image](https://user-images.githubusercontent.com/8946502/107735380-ab626e80-6d53-11eb-9f2c-3b704d877edb.png)

I assume `check-full` is what should be running given that's the `test` script in `package.json`? Not sure though. 

Also, `check-full` spams the console with outputs of the holidays one per line, which means that all the previous results about actually testing the library get pushed out of my terminal (can't scroll back to them, it only goes back so far).

ypid commented:
The file `locales/word_error_correction.yaml` got added in the rebase, probably because of conflicts. Fixed that.

I should also have addressed your other findings, thanks. So now I would just like someone else with up-to-date JavaScript experience to have a look. @KristjanESPERANTO can you take a quick look on this?

KristjanESPERANTO commented:
I would like to try to help. I took the folders branch from @napei and was able to do a build.

Unfortunately, I can't get the `opening_hours.js` to run in a browser and get the following error message:

`Uncaught TypeError: i18n__default.default is undefined`

I tried to replace the deprecated i18next-client [1] with the i18next module, but I still had no success.

Did I understand correctly that there will be no more `opening_hours+deps(min).js`?

[1] https://www.npmjs.com/package/i18next-client

napei commented:
> Unfortunately, I can't get the `opening_hours.js` to run in a browser and get the following error message:
> 
> `Uncaught TypeError: i18n__default.default is undefined`

Haven't tried in a browser but `npm run build` will output a `opening_hours+deps.min.js` as expected. That error is expected as it's the version without dependencies i.e. `i18next-client`

You can try `make clean` then build again?

KristjanESPERANTO commented:
> You can try `make clean` then build again?

I guess I did something wrong. Now I've got an `opening_hours+deps.min.js` 🙂 and it's working in a browser 👍

napei commented:
> Also, check-full spams the console with outputs of the holidays one per line, which means that all the previous results about actually testing the library get pushed out of my terminal (can't scroll back to them, it only goes back so far).

@ypid thoughts on this?

<details>
<summary>Output of tests (note I can't scroll back to look at the other test results)</summary>

```text
2021-10-04 Курбан-байрам
2021-10-11 День Республики Башкирии
2021-11-04 День народного единства
2021-12-24 День Конституции Башкортостана
ru_bryansk:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-04-29 Радоница
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-09-17 День освобождения города Брянска
2021-11-04 День народного единства
ru_buryatia:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-01-14 Сагаалган
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-11-04 День народного единства
ru_chechnya:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-04-16 День мира в Чеченской Республике
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-07-28 Ураза-байрам
2021-10-04 Курбан-байрам
2021-11-04 День народного единства
ru_chuvashia:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-06-24 День Чувашской республики
2021-11-04 День народного единства
ru_dagestan:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-07-26 День Конституции Республики Дагестан
2021-07-28 Ураза-байрам
2021-09-15 День единства народов Дагестана
2021-10-04 Курбан-байрам
2021-11-04 День народного единства
ru_ingushetia:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-04 День образования Республики Ингушетия
2021-06-12 День России
2021-07-28 Ураза-байрам
2021-10-04 Курбан-байрам
2021-11-04 День народного единства
ru_kabardino_balkaria:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-03-28 День возрождения балкарского народа
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-05-21 Черкесский день траура
2021-06-12 День России
2021-07-28 Ураза-байрам
2021-09-01 День государственности Кабардино-Балкарской Республики
2021-10-04 Курбан-байрам
2021-11-04 День народного единства
ru_kalmykia:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-01-14 Цаган Сар
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-04-05 День принятия Степного Уложения (Конституции) Республики Калмыкия
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-06 День рождения Будды Шакьямун
2021-06-12 День России
2021-11-04 День народного единства
2021-12-15 Зул
2021-12-28 День памяти жертв депортации калмыцкого народа
ru_karachay_cherkess:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-03 День возрождения карачаевского народа
2021-05-09 День Победы
2021-06-12 День России
2021-07-28 Ураза-байрам
2021-10-04 Курбан-байрам
2021-11-04 День народного единства
ru_karelia:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-08 День Республики Карелия
2021-06-12 День России
2021-09-30 День освобождения Карелии от фашистских захватчиков
2021-11-04 День народного единства
ru_komi:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-08-22 День Республики Коми
2021-11-04 День народного единства
ru_sakha:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-04-27 День Республики Саха
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-06-23 Ысыах
2021-09-27 День государственности Республики Саха
2021-11-04 День народного единства
ru_saratov:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-04-29 Радоница
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-11-04 День народного единства
ru_tatarstan:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-07-28 Ураза-байрам
2021-08-30 День Республики Татарстан
2021-10-04 Курбан-байрам
2021-11-04 День народного единства
2021-11-06 День Конституции Республики Татарстан
ru_tuva:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-01-14 Народный праздник Шагаа
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-08-15 День Республики Тыва
2021-11-04 День народного единства
ru_udmurtia:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-05-31 День Государственности Удмуртской Республики
2021-06-12 День России
2021-11-04 День народного единства
ru:
2021-01-01 1. Новогодние каникулы
2021-01-02 2. Новогодние каникулы
2021-01-03 3. Новогодние каникулы
2021-01-04 4. Новогодние каникулы
2021-01-05 5. Новогодние каникулы
2021-01-06 6. Новогодние каникулы
2021-01-07 Рождество Христово
2021-01-08 8. Новогодние каникулы
2021-02-23 День защитника Отечества
2021-03-08 Международный женский день
2021-05-01 Праздник Весны и Труда
2021-05-09 День Победы
2021-06-12 День России
2021-11-04 День народного единства
se:
2021-01-01 nyårsdagen
2021-01-06 trettondedag jul
2021-04-02 långfredagen
2021-04-04 påskdagen
2021-04-05 annandag påsk
2021-05-01 första maj
2021-05-23 pingstdagen
2021-06-06 nationaldagen
2021-06-26 midsommardagen
2021-11-06 alla helgons dag
2021-12-25 juldagen
2021-12-26 annandag jul
sk:
2021-01-01 Deň vzniku Slovenskej republiky
2021-01-06 Zjavenie Pána
2021-04-02 Veľký piatok
2021-04-04 Veľkonočná nedeľa
2021-04-05 Veľkonočný pondelok
2021-05-01 Sviatok práce
2021-05-08 Deň víťazstva nad fašizmom
2021-07-05 Sviatok svätého Cyrila a Metoda
2021-08-29 Výročie Slovenského národného povstania
2021-09-01 Deň Ústavy Slovenskej republiky
2021-09-15 Sviatok Panny Márie Sedembolestnej
2021-11-01 Sviatok všetkých svätých
2021-11-17 Deň boja za slobodu a demokraciu
2021-12-24 Štedrý deň
2021-12-25 Prvý sviatok vianočný
2021-12-26 Druhý sviatok vianočný
us_ak:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-03-29 Seward's Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-18 Alaska Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_al:
2021-01-01 New Year's Day
2021-01-18 Robert E. Lee/Martin Luther King Birthday
2021-02-15 George Washington/Thomas Jefferson Birthday
2021-04-26 Confederate Memorial Day
2021-05-31 Memorial Day
2021-06-07 Jefferson Davis' Birthday
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ar:
2021-01-01 New Year's Day
2021-01-18 Dr. Martin Luther King Jr. and Robert E. Lee's Birthdays
2021-02-15 George Washington's Birthday and Daisy Gatson Bates Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
us_az:
2021-01-01 New Year's Day
2021-01-18 Dr. Martin Luther King Jr./Civil Rights Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ca:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-03-31 César Chávez Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_co:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ct:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-12 Lincoln's Birthday
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_dc:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-16 Emancipation Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_de:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Day After Thanksgiving
2021-12-25 Christmas Day
us_fl:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Friday after Thanksgiving
2021-12-25 Christmas Day
us_ga:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-04-26 Confederate Memorial Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Robert E. Lee's Birthday
2021-12-24 Washington's Birthday
2021-12-25 Christmas Day
us_gu:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-03-05 Guam Discovery Day
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-07-21 Liberation Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 All Souls' Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-08 Lady of Camarin Day
2021-12-25 Christmas Day
us_hi:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-03-26 Prince Jonah Kuhio Kalanianaole Day
2021-05-31 Memorial Day
2021-06-11 Kamehameha Day
2021-07-04 Independence Day
2021-08-20 Statehood Day
2021-09-06 Labor Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ia:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-12 Lincoln's Birthday
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_id:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr.-Idaho Human Rights Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_il:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-12 Lincoln's Birthday
2021-02-15 Washington's Birthday
2021-03-01 Casimir Pulaski Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_in:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-04 Primary Election Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Lincoln's Birthday
2021-12-25 Christmas Day
us_ks:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ky:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
2021-12-31 New Year's Eve
us_la:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-02-16 Mardi Gras
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ma:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-19 Patriots' Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_md:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Native American Heritage Day
2021-12-25 Christmas Day
us_me:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-19 Patriots' Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_mi:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
2021-12-31 New Year's Eve
us_mn:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_mo:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-08 Truman Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ms:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King's and Robert E. Lee's Birthdays
2021-02-15 Washington's Birthday
2021-04-26 Confederate Memorial Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_mt:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
2021-12-31 New Year's Eve
us_nc:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Day after Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
2021-12-26 Day after Christmas
us_nd:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ne:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-30 Arbor Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_nh:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Civil Rights Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Day after Thanksgiving
2021-12-25 Christmas Day
us_nj:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-12 Lincoln's Birthday
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_nm:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Day after Thanksgiving
2021-12-25 Christmas Day
us_nv:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-29 Nevada Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Family Day
2021-12-25 Christmas Day
us_ny:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-12 Lincoln's Birthday
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_oh:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_ok:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Day after Thanksgiving
2021-12-25 Christmas Day
us_or:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_pa:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-06-14 Flag Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_pr:
2021-01-01 Día de Año Nuevo
2021-01-06 Día de Reyes
2021-01-11 Natalicio de Eugenio María de Hostos
2021-01-18 Natalicio de Martin Luther King, Jr.
2021-02-15 Día de los Presidentes
2021-03-22 Día de la Abolición de Esclavitud
2021-04-02 Viernes Santo
2021-04-19 Natalicio de José de Diego
2021-05-31 Recordación de los Muertos de la Guerra
2021-07-04 Día de la Independencia
2021-07-25 Constitución de Puerto Rico
2021-07-27 Natalicio de Dr. José Celso Barbosa
2021-09-06 Día del Trabajo
2021-10-11 Día de la Raza Descubrimiento de América
2021-11-11 Día del Veterano
2021-11-19 Día del Descubrimiento de Puerto Rico
2021-11-25 Día de Acción de Gracias
2021-12-24 Noche Buena
2021-12-25 Día de Navidad
us_ri:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-08-09 Victory Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_sc:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-10 Confederate Memorial Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_sd:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Native American Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_tn:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-04-02 Good Friday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
us_tx:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Friday after Thanksgiving
2021-12-24 Christmas Eve
2021-12-25 Christmas Day
2021-12-26 Day after Christmas
us_ut:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-07-24 Pioneer Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_va:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_vi:
2021-01-01 New Year's Day
2021-01-06 Three Kings Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-03-31 Transfer Day
2021-04-01 Holy Thursday
2021-04-02 Good Friday
2021-04-05 Easter Monday
2021-05-31 Memorial Day
2021-07-03 Emancipation Day
2021-07-04 Independence Day
2021-07-26 Hurricane Supplication Day
2021-09-06 Labor Day
2021-10-11 Virgin Islands-Puerto Rico Friendship Day
2021-10-25 Hurricane Thanksgiving
2021-11-01 Liberty Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
2021-12-26 Christmas Second Day
2021-12-31 New Year's Eve
us_vt:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-03-02 Town Meeting Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-08-16 Battle of Bennington
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_wa:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_wi:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-08-10 Primary Election Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-02 Election Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us_wv:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-06-20 West Virginia Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-11-26 Lincoln's Day
2021-12-25 Christmas Day
us_wy:
2021-01-01 New Year's Day
2021-01-18 Martin Luther King, Jr. Day
2021-02-15 Washington's Birthday
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-10-11 Columbus Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day
us:
2021-01-01 New Year's Day
2021-05-31 Memorial Day
2021-07-04 Independence Day
2021-09-06 Labor Day
2021-11-11 Veterans Day
2021-11-25 Thanksgiving
2021-12-25 Christmas Day

$>  
```
</details>

ypid commented:
Thanks for testing.

> @ypid thoughts on this?

During normal development (no big refactor) `make check` should be enough. `make check-full` runs during CI and allows anyone to review the contributed definitions. You can use `less` locally so that you can still read the beginning.

> I think it may be better for the end user to have these dependencies [...] installed as devDependencies

Works for me. Can you change it?

napei commented:
> Works for me. Can you change it?

@ypid done. I've removed the `dependencies-user-wide-get` make script and updated the github action to reflect. I took the liberty of adding caching to CI as well to speed things up. Double checked and all relevant scripts still run just fine and tests pass as does CI.

Not sure if you're still using travis but updated that anyway.

ypid commented:
Merged, thanks. I removed the left over `.travis.yml` file, thanks for the hint. I forgot that when migrating.
